### PR TITLE
week-4 is completed

### PR DESCRIPTION
**Pull request by OlenaZh**
# Submitting Week 4 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -16,35 +18,237 @@
...
+//return arr
+
+function shuffleArray(arr) {
+    arr.shift()
+    for (let i = arr.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [arr[i], arr[j]] = [arr[j], arr[i]];
+    }
+return arr
+}
```
Comment (elysemontano) [inline code-challenges.test.js:83]: 👏 Nice work!

File: code-challenges.test.js [hunk 2]
```diff
@@ -16,35 +18,237 @@
...
 
 // b) Create the function that makes the test pass.
+
+//declare a function and name it nonoDuplicateArr
+//pass 2 parameters (arr1, arr2)
+//use concat() method to combine two arrays into one
+//use set() method to remove duplicats from the rray. 
+    //1-st -> convert an array of duplicates to a Set. The new Set will implicitly remove duplicates elements.
+    //2-nd -> convert the set back to an array
+//return new, modified array (uniqueArr)
```
Comment (elysemontano) [inline code-challenges.test.js:234]: Great work!

File: interview-questions.md [hunk 3]
```diff
@@ -9,41 +9,79 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 ## Looking Ahead: Terms for Next Week
 
-1. PostgreSQL:
+1. PostgreSQL: free and open-source relational database management system.
 
-2. Ruby on Rails:
+2. Ruby on Rails: is a server-side web app framework written in Ruby under the MIT License. 
 
-3. ORM:
+3. ORM:object-relational mapping
```
Comment (elysemontano) [inline interview-questions.md:83]: Looks like you are missing numbers 4 and 5 in the looking ahead section.

File: ruby-challenges.rb [hunk 4]
```diff
@@ -4,31 +4,62 @@
...
 num1 = 7
-# Expected output: '7 is odd'
 num2 = 42
-# Expected output: '42 is even'
 num3 = 221
-# Expected output: '221 is odd'
 
 
+def even_or_odd num
+    if num % 2 == 0
```
Comment (elysemontano) [inline ruby-challenges.rb:12]: Great work on your logic!  Ruby also has a built-in method called .even? that does some of this work for you as well.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Request Changes: It looks like you missed two problems in the looking ahead section in interview questions.  Please complete those two and I will approve this PR.
Other comments:
Overall excellent work Olena!  Logic and tests are consistently good.  There are times when your pseudocode gets a little thin, so please try to keep your pseudocode thorough on all problems.

File: interview-questions.md (hunk 3 above)
Comment (OlenaZh) [reply interview-questions.md:83]: Hi  Elyse,
I have completed 4 and 5 in the Looking Ahead section. 
In addition, I have installed Code Spell Checker and I love it. Now my assessment looks cleaner. Thank you for the recommendation.

Comment (elysemontano) [reply interview-questions.md:83]: Nice work!

**Review by elysemontano — APPROVED** [review]

**Pull request merged**